### PR TITLE
Add events for major lifecycle events

### PR DESCRIPTION
**Pull request by bowei**
{Add,Remove}Nodes from InstanceGroup
SyncIngress, TranslateIngress, IPChanged, GarbageCollection

### CODE REVIEW

**Comment by bowei** [comment]
/assign @rramkumar1 
cc: @spencerhance

File: pkg/events/events_test.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package events
+
+import (
+	"testing"
+)
+
+func TestTruncatedStringList(t *testing.T) {
```
Comment (rramkumar1) [inline pkg/events/events_test.go:7]: address TODO

File: pkg/events/events.go [hunk 2]
```diff
@@ -17,9 +17,22 @@ limitations under the License.
 package events
 
 import (
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/tools/record"
 )
 
+const (
+	AddNodes    = "IngressGCE_AddNodes"
+	RemoveNodes = "IngressGCE_RemoveNodes"
```
Comment (rramkumar1) [inline pkg/events/events.go:26]: Why do we need the IngressGCE prefix? Is it to distinguish from other Ingress controllers for the same type of action?

File: pkg/events/events.go [hunk 3]
```diff
@@ -17,9 +17,22 @@ limitations under the License.
...
+
+	SyncIngress       = "Sync"
+	TranslateIngress  = "Translate"
+	IPChanged         = "IPChanged"
+	GarbageCollection = "GarbageCollection"
+
+	SyncService = "Sync"
```
Comment (rramkumar1) [inline pkg/events/events.go:33]: what is the difference between this and SyncIngress?

Comment (bowei) [reply pkg/events/events.go:33]: the value is not the same, but this identifies it so we can easily change it if needed

**Comment by bowei** [comment]
@rramkumar1 PTAL

**Pull request merged**